prow/github/client.go

```diff
@@ -387,7 +394,16 @@ func (c *Client) doRequest(method, path, accept string, body interface{}) (*http
 	if err != nil {
 		return nil, err
 	}
-	req.Header.Set("Authorization", "Token "+c.token)
+
+	var token string
+	if c.secretAgent != nil {
```

Comment:
When is it valid for the agent to be `nil`? If it's for testing, let's not allow that and use a fake agent

Reply:
For start, only hook is using the secretAgent, so the rest are using the token as they used to. That's why I am giving nil secretAgent from the rest binaries. This should be removed when all the binaries will use the secretAgent.

Reply:
Let's get them all migrated in one pull, please